Source/UI/PaymentForm/ViewModel/DefaultPaymentViewModel.swift

```diff
@@ -154,14 +154,15 @@ extension DefaultPaymentViewModel: PaymentViewControllerDelegate {
     validateMandatoryInputProvided()
   }
 
-  func payButtonIsPressed() {
-    guard let card = cardDetails.getCard() else { return }
-    isLoading = true
-    checkoutAPIService.createToken(.card(card)) { [weak self] result in
-      self?.isLoading = false
-      self?.cardTokenRequested?(result)
+    func payButtonIsPressed() {
+        guard let card = cardDetails.getCard() else { return }
```

Comment:
Unrelated to PR , Question : What happens if cardDetails.getCard() is nil, will this do a silent return. Or this will never happen.

Reply:
As long as the button enabling / disabling is synced correctly it should not happen. But if it did, it would just silently fail. This could actually be logged as a `warn`

Reply:
As long as the button enabling / disabling is synced correctly it should not happen. But if it did, it would just silently fail. 

This could actually be logged as a `warn`? I think the warn comes 2 PRs later, so if you're keen on reporting that, I'll open a follow up after all other events have their PRs open.

Reply:
As warn comes in next PR (#316 ), I've opened a PR for this after all the other logging work (#320 )